cairo_kernel/repl.py

```diff
@@ -84,12 +86,21 @@ def __init__(self):
 
     def parse(self, code: str) -> Union[CodeElement, Expression]:
         try:
-            return parse(
+            first_parse = parse(
                 filename=INPUT_FILENAME,
                 code=code,
                 code_type="expr",
                 expected_type=Expression,
             )
+            if isinstance(first_parse, ExprFuncCall):
```

Comment:
interesting, I think we need to properly handle parsing of code snippets there are a lot of edge cases

Reply:
Agreed it's not the most elegant solution, but if it works it avoids having to deeply look at the parsers' logic (ebnf rules etc..)